### PR TITLE
Bug 1358601 - Add regexes and names for new DevEdition jobs.

### PR DESCRIPTION
**Pull request by mozbhearsum**
We'll be turning on DevEdition builds and tests on mozilla-beta shortly (builds tomorrow, tests next week). These builds will be Firefox builds, but done with a different build configuration. We will only be doing opt builds and tests. I *think* this patch covers all of those bases.

More context for this in https://bugzilla.mozilla.org/show_bug.cgi?id=1358601.

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: ui/js/values.js [hunk 1]
```diff
@@ -87,6 +87,15 @@ treeherder.value("thPlatformMap", {
...
 
+    "linux32-devedition": "Linux DevEdition",
+    "linux64-devedition": "Linux x64 DevEdition",
+    "osx-10-7-devedition": "OS X 10.7 DevEdition",
+    "osx-10-10-devedition": "OS X 10.10 DevEdition",
+    "windowsxp-devedition": "Windows XP DevEdition",
+    "windows7-32-devedition": "Windows 7 DevEdition",
+    "windows7-32-vm-devedition": "Windows 7 VM DevEdition",
+    "windows8-64-devedition": "Windows 8 x64 DevEdition",
+
```
Comment (KWierso) [inline ui/js/values.js]: Putting these here means these platforms will show near the very bottom of the list on each push. If you move them up intermingled with the other desktop platforms (lines 4-36), they'll show up next to those platforms.

Comment (mozbhearsum) [inline ui/js/values.js]: Ah, sure! Will do.

File: tests/etl/test_buildbot.py [hunk 2]
```diff
@@ -843,6 +843,15 @@
       'platform': {'arch': 'x86_64',
                    'os': 'linux',
                    'os_platform': 'linux64-stylo'}}),
+    ('Ubuntu HW 12.04 x64 devedition mozilla-beta talos g3',
+     {'build_type': 'opt',
+      'name': {'group_name': 'Talos Performance',
+               'group_symbol': 'T-deved',
```
Comment (edmorley) [inline tests/etl/test_buildbot.py]: This should be `T` to make Travis pass.

**Comment by mozbhearsum** [comment]
I'll fix up the test symbol. I think I was originally looking at putting these on existing lines, so I thought I might need that to be different...

**Pull request merged**